### ISSUE TITLE
xmlrpc submodule is not ready for python3

### ISSUE THREAD

**Issue by Allda** [release-engineering/kobo#64]
Class SafeCookieTransport() which is in xmlrpc submodule uses old (2.7) interface of xmlrpclib library. 
Following code snippet shows the bug:

```python
import xmlrpc.client
from kobo.xmlrpc import SafeCookieTransport

url = 'https://errata.devel.redhat.com/errata/xmlrpc.cgi'

xmlrpc = xmlrpc.client.ServerProxy(url, transport=SafeCookieTransport())

print(xmlrpc.get_advisory_cdn_file_list('RHSA-2018:2942-03'))
```
Here is the output:
```
Traceback (most recent call last):
  File "xml_rpc_test.py", line 9, in <module>
    print(xmlrpc.get_advisory_cdn_file_list('RHSA-2018:2942-03'))
  File "/usr/lib64/python3.6/xmlrpc/client.py", line 1112, in __call__
    return self.__send(self.__name, args)
  File "/usr/lib64/python3.6/xmlrpc/client.py", line 1452, in __request
    verbose=self.__verbose
  File "/usr/lib64/python3.6/xmlrpc/client.py", line 1154, in request
    return self.single_request(host, handler, request_body, verbose)
  File "/home/araszka/.local/lib/python3.6/site-packages/kobo/xmlrpc.py", line 450, in _single_request
    self.send_request(h, handler, request_body)
  File "/home/araszka/.local/lib/python3.6/site-packages/kobo/xmlrpc.py", line 557, in send_request
    return xmlrpclib.SafeTransport.send_request(self, connection, handler, request_body)
TypeError: send_request() missing 1 required positional argument: 'debug'
```
New version of xmlrpc library for python 3 and its interface have been updated and 'debug' argument has been added to send_request() function.
https://github.com/python/cpython/blob/3.7/Lib/xmlrpc/client.py#L1266

**Comment by yontalcar** [comment]
Hi, is there any update on this issue? I'll take a look if nobody is working on it.

**Comment by alexandrevicenzi** [comment]
@yontalcar no updates so far, kobo is not py3 ready yet.

**Comment by kdudka** [comment]
I thought it was implemented in PR #48.  Is the current status of py3 migration tracked anywhere?

**Comment by rohanpm** [comment]
PR #48 got as far as making it possible to install kobo with python3 and adding python3 coverage in the CI.  It didn't fix all bugs with python3.  There are still several known and probably many unknown.

"git grep xfail tests" gives a more complete picture of the python3 status, you can find many tests currently marked as expected failures on python3.
